xep-0234.xml

```diff
@@ -354,7 +360,12 @@
     <tr>
       <td>hash</td>
       <td>A hash of the file content, using the &lt;hash/&gt; element defined in &xep0300; and qualifed by the 'urn:xmpp:hashes:2' namespace. Multiple hashes MAY be included for hash agility.</td>
-      <td>REQUIRED when offering a file, otherwise OPTIONAL</td>
+      <td>See &lt;hash-used/&gt;</td>
+    </tr>
+    <tr>
+      <td>hash-used</td>
+      <td>Alternatively to a &lt;hash/&gt; element, the initiator can also include a &lt;hash-used/&gt; element. This comes handy to avoid reading the file twice to calculate the hash.</td>
```

Comment:
> it seems that the "hash-used" element is sent in addition to,

Just to clarify: The idea is that it is **not** send in addition to. You either send a hash element, which includes the hash algo and the hash value, OR you send a hash-used, which just includes the hash algo.